Explainer/PowerPointReader.py

```diff
@@ -0,0 +1,164 @@
+from pptx import Presentation
+from GPTExplainer import sendToChatGPT  # Import the sendToChatGPT function
+import os
+
+
+# The different layouts
+TITLE = 0
+TITLE_AND_CONTENT = 1
+SECTION_HEADER = 2
+TWO_CONTENT = 3
+COMPARISON = 4
+TITLE_ONLY = 5
+BLANK = 6
+CONTENT_WITH_CAPTION = 7
+PICTURE_WITH_CAPTION = 8
+
+'''
+extract the text from the slide
+'''
+
+
+
+def extract_text(slide):
+    text = ""
+    for shape in slide.shapes:
+        if shape != slide.shapes.title:
+            text += get_shape_info(shape)
+    return text
+
+
+'''
+get's the text from the specific shape
+'''
+
+
+def get_shape_info(shape):
+    txt = ""
+    if shape.has_text_frame:
+        for paragraph in shape.text_frame.paragraphs:
+            for run in paragraph.runs:
+                txt += run.text
+    return txt
+
+
+'''
+extract extra notes from the slide
+'''
+
+
+def extract_notes(slide):
+    notes_txt = ""
+    if slide.has_notes_slide:
+        notes_slide = slide.notes_slide
+        # Access the notes content
+        notes_text = ""
+        for shape in notes_slide.shapes:
+            notes_text += get_shape_info(shape)
+    return notes_txt
+
+
+
+'''
+gets extra information about the powerpoint from the useer
+'''
+
+
+def readExtraInfo(pPname):
+    extra_info = {}
+    extra_info['age'] = input("What age group is the PowerPoint designed for?")
+    extra_info['lesson'] = input("What course is the PowerPoint for?")
+    extra_info['subject'] = input("What is the subject of the lesson")
+    extra_info['pPname'] = pPname
+    extra_info['info'] = Add_Info_Slide()
+    return extra_info
+
+
+
+'''
+returns a dictionary with extra information for certain slides
+'''
+
+def Add_Info_Slide():
```

Comment:
Fix everywhere

Reply:
"This name does not adhere to python's naming conventions. Names should be indicative, and when used properly can reduce redundant documentation.
In addition, each Variable, function, module, constant and class names has a specific naming convention. 

Read this link to learn the proper conventions - 
https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html

If you don't know what CamelCase and snake_case means, read this link - 
https://www.google.com/amp/s/www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference%3famp=1

For a general overview on python conventions (PEP-8), read this link - 
https://peps.python.org/pep-0008/
"